spec/message.abnf

```diff
@@ -52,16 +55,19 @@ unquoted-start = name-start / DIGIT / "."
                / %xB7 / %x300-36F / %x203F-2040
 
 
-; reserve sigils for private-use by implementations
-private-use    = private-start reserved-body
-private-start  = "^" / "&"
+; Reserve additional .keywords for use by future versions of this specification.
+reserved-statement = reserved-keyword [s [reserved-body [s]]] 1*expression
+; Note that the following expression is a simplification,
+; as this rule MUST NOT be considered to match existing keywords.
+; As regexp, this would be: /\.(?!input|local|match)[a-z]+/
+reserved-keyword = "." 1*(%x61-7A)
 
-; reserve additional sigils for use by 
-; future versions of this specification
-reserved       = reserved-start reserved-body
-reserved-start = "!" / "@" / "#" / "%" / "*" / "<" / ">" / "/" / "?" / "~"
-reserved-body  = *( [s] 1*(reserved-char / reserved-escape / quoted))
+; Reserve additional sigils for use by future versions of this specification.
+reserved-annotation = reserved-annotation-start reserved-body
+reserved-annotation-start = "!" / "@" / "#" / "%" / "*"
+                          / "<" / ">" / "/" / "?" / "~"
 
+reserved-body  = *([s] 1*(reserved-char / reserved-escape / quoted))
 reserved-char  = %x00-08        ; omit HTAB and LF
                / %x0B-0C        ; omit CR
                / %x0E-19        ; omit SP
```

Comment:
`reserved-char` should omit `.` so e.g. `.new {$a} .hotness {$b}` is recognized as two distinct reserved statements and `.prefix {$foo} .local ${var} = ${|val|}` is recognized as a reserved statement followed by a local declaration, while still allowing reserved statements to include non-consecutive expressions (like `local-declaration = local s variable [s] "=" [s] expression` already does).
```suggestion
reserved-char  = %x00-08        ; omit HTAB and LF
               / %x0B-0C        ; omit CR
               / %x0E-19        ; omit SP
               / %x21-2D        ; omit .
               / %x2F-5B        ; omit \
```

And accordingly, `reserved-escape` should be generalized to e.g. `reserved-statement-escape = backslash ( %x00-D7FF / %xE000-10FFFF )`.

Reply:
Note that this precludes syntax like that used now in `local-declaration`, where we have a `variable` `name` at the top level, and `name` _can_ include a `.`. Is that intentional?

Reply:
No, it will need further tweaking to match `.` only after `name-start` or `name-char`, as inside `name`.